dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -679,64 +681,87 @@ private void deleteTreesForInodes(List<String> inodes) throws DotDataException {
     }
 
 	@Override
-	protected int deleteOldContent(Date deleteFrom) throws DotDataException {
-	    ContentletCache cc = CacheLocator.getContentletCache();
-        Calendar calendar = Calendar.getInstance();
+    protected int deleteOldContent(final Date deleteFrom) throws DotDataException {
+        final Calendar calendar = Calendar.getInstance();
         calendar.setTime(deleteFrom);
         calendar.set(Calendar.HOUR_OF_DAY, 0);
         calendar.set(Calendar.MINUTE, 0);
         calendar.set(Calendar.SECOND, 0);
         calendar.set(Calendar.MILLISECOND, 0);
-        Date date = calendar.getTime();
-        //Because of the way Oracle databases handle dates,
-        //this string is converted to Uppercase.This does
-        //not cause a problem with the other databases
-        DotConnect dc = new DotConnect();
+        final Date date = calendar.getTime();
 
-        String countSQL = ("select count(*) as count from contentlet");
+        DotConnect dc = new DotConnect();
+        final String countSQL = "select count(*) as count from contentlet";
         dc.setSQL(countSQL);
         List<Map<String, String>> result = dc.loadResults();
-        int before = Integer.parseInt(result.get(0).get("count"));
-
-        String deleteContentletSQL = "delete from contentlet where identifier<>'SYSTEM_HOST' and mod_date < ? " +
-        "and not exists (select * from contentlet_version_info where working_inode=contentlet.inode or live_inode=contentlet.inode)";
-        dc.setSQL(deleteContentletSQL);
-        dc.addParam(date);
-        dc.loadResult();
-
-        String deleteOrphanInodes="delete from inode where type='contentlet' and idate < ? and inode not in (select inode from contentlet)";
-        dc.setSQL(deleteOrphanInodes);
+        final int before = Integer.parseInt(result.get(0).get("count"));
+        dc = new DotConnect();
+        dc.setSQL("SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < ? AND NOT EXISTS " +
+                "(SELECT working_inode, live_inode FROM contentlet_version_info WHERE working_inode = contentlet.inode OR " +
+                "live_inode = contentlet.inode)");
```

Comment:
Just by separating this into two queries and using UNION on them you will probably get a high performance gain:

Original query
```
EXPLAIN ANALYSE SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < now() AND NOT EXISTS
(SELECT working_inode, live_inode FROM contentlet_version_info WHERE working_inode = contentlet.inode OR
                live_inode = contentlet.inode)
```
Proposed query - One query for live, one for working and UNION
```
EXPLAIN ANALYSE SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < now() AND NOT EXISTS
    (SELECT working_inode FROM contentlet_version_info WHERE working_inode = contentlet.inode)
UNION
        SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < now() AND NOT EXISTS
    (SELECT live_inode FROM contentlet_version_info WHERE live_inode = contentlet.inode)```